### PR TITLE
Numba function operator, docs update

### PR DESCRIPTION
**Pull request by majra20**
Signed-off-by: Rafal Maj <rmaj@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It fixes a bug *bug description*
- It adds new feature needed because of *why we need this feature*
- Refactoring to improve *what*

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *[ Explain solution of the problem, a new feature added here. ]*
 - Affected modules and functionalities:
     *[ Describe here what was changed, added, removed. ]*
 - Key points relevant for the review:
     *[ Describe here what is the most important part that reviewers should focus on. ]*
 - Validation and testing:
     *[ Describe here if and how this PR is tested. ]*
 - Documentation (including examples):
     *[ Describe here if documentation and examples were updated. ]*

**JIRA TASK**: *[Use DALI-XXXX or NA]*

### CODE REVIEW

File: dali/operators/numba_function/numba_func.cc [hunk 1]
```diff
@@ -112,7 +112,8 @@ This function must work in Numba ``nopython`` mode.)code", DALI_PYTHON_OBJECT)
 This function is invoked once per batch. Also this function must work in Numba ``nopython`` mode.)code",
                   DALI_PYTHON_OBJECT, nullptr)
   .AddOptionalArg("batch_processing", R"code(Determines whether the function is invoked once per batch or
-separately for every sample in the batch.)code", false);
+separately for every sample in the batch. For CPU usage we recommend not using this option. When operator
```
Comment (JanuszL) [inline dali/operators/numba_function/numba_func.cc:115]:
```suggestion
separately for every sample in the batch.

For CPU usage we recommend not using this option. When operator 
```

**Review by JanuszL — APPROVED** [review]

File: dali/operators/numba_function/numba_func.cc [hunk 2]
```diff
@@ -112,7 +112,10 @@ This function must work in Numba ``nopython`` mode.)code", DALI_PYTHON_OBJECT)
 This function is invoked once per batch. Also this function must work in Numba ``nopython`` mode.)code",
                   DALI_PYTHON_OBJECT, nullptr)
   .AddOptionalArg("batch_processing", R"code(Determines whether the function is invoked once per batch or
-separately for every sample in the batch.)code", false);
+separately for every sample in the batch.
```
Comment (mzient) [inline dali/operators/numba_function/numba_func.cc:115]:
```suggestion
separately for each sample in the batch.
```

**Comment by majra20** [comment]
!build

File: dali/operators/numba_function/numba_func.cc [hunk 3]
```diff
@@ -112,7 +112,10 @@ This function must work in Numba ``nopython`` mode.)code", DALI_PYTHON_OBJECT)
...
+
+The best performance for the CPU is obtained when this option is turned off. When operator 
+is working in per-sample basis multithreading is utilized, which may increase perfomance.)code", false);
```
Comment (mzient) [inline dali/operators/numba_function/numba_func.cc:118]: When ``batch_processing`` is set to ``True``, the function processes the whole batch. It is necessary if the function has to perform cross-sample operations and may be beneficial if significant part of the work can be reused. For other use cases, specifying ``False`` and using per-sample processing function allows the operator to process samples in parallel.

File: dali/operators/numba_function/numba_func.cc [hunk 4]
```diff
@@ -131,7 +134,7 @@ DALI_SCHEMA(NumbaFuncImpl)
   .AddOptionalArg<int>("setup_fn", R"code(Address of setup function setting shapes for outputs.
 This function is invoked once per batch.)code", 0)
   .AddOptionalArg("batch_processing", R"code(Determines whether the function is invoked once per batch or
-separately for every sample in the batch.)code", false);
+separately for each sample in the batch.)code", false);
```
Comment (mzient) [inline dali/operators/numba_function/numba_func.cc:137]: Shouldn't the explanation be here, too?

Comment (jantonguirao) [reply dali/operators/numba_function/numba_func.cc:137]: Perhaps it should be here only.

Comment (majra20) [reply dali/operators/numba_function/numba_func.cc:137]: This isn't shown in our documentation, it has `MakeInternal()` option.

**Comment by majra20** [comment]
!build

**Review by jantonguirao — APPROVED** [review]

**Comment by majra20** [comment]
!build

**Pull request merged**